### PR TITLE
[tests-only] Fix path encoding in tests steps

### PR DESCRIPTION
**Pull request by saw-jan**
## Related Issue
- https://github.com/owncloud/core/issues/35741

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/WebDav.php [hunk 1]
```diff
@@ -3556,7 +3557,11 @@ private function deleteUpload($user, $id, $headers) {
 	 */
 	public function encodePath($path) {
 		// slashes need to stay
-		return \str_replace('%2F', '/', \rawurlencode($path));
+		$encodedPath = \str_replace('%2F', '/', \rawurlencode($path));
+		// do not encode '(' and ')'
+		$encodedPath = \str_replace('%28', '(', $encodedPath);
+		$encodedPath = \str_replace('%29', ')', $encodedPath);
```
Comment (dpakach) [inline tests/acceptance/features/bootstrap/WebDav.php:3563]: why is this needed? dont we need to encode these characters in the URL?

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/WebDav.php:3563]: Yeah, `()` should not be encoded but `rawurlencode` encodes it.

**Review by dpakach — COMMENTED** [review]
lgtm 👍🏼

**Comment by kiranparajuli589** [comment]
lgtm 👍🏼

File: tests/acceptance/features/bootstrap/WebDav.php (hunk 1 above)
Comment (phil-davis) [reply tests/acceptance/features/bootstrap/WebDav.php:3563]: yes, https://www.php.net/manual/en/function.rawurlencode.php does more than what we need here. It does not have options to just encode a subset of special characters. At least the special exceptions are in a common method here `encodePath` so that everything else can call it consistently.

**Review by phil-davis — APPROVED** [review]
